factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/listen/PgListener.java

```diff
@@ -57,85 +58,141 @@
 
-    private void listen() {
+    @VisibleForTesting
+    protected void listen() {
         log.trace("Starting instance Listener");
-        CountDownLatch l = new CountDownLatch(1);
-        listenerThread = new Thread(() -> {
+        listenerThread = new Thread(new NotificationReceiverLoop(), "PG Instance Listener");
+        listenerThread.setDaemon(true);
+        listenerThread.setUncaughtExceptionHandler(
+                (t, e) -> log.error("thread " + t + " encountered an unhandled exception", e));
+        listenerThread.start();
+        try {
+            log.info("Waiting to establish postgres listener (max 15sec.)");
+            boolean await = countDownLatch.await(15, TimeUnit.SECONDS);
+            log.info("postgres listener " + (await ? "" : "not ") + "established");
+        } catch (InterruptedException ignored) {
+        }
+    }
+
+    @VisibleForTesting
+    protected class NotificationReceiverLoop implements Runnable {
+
+        @Override
+        public void run() {
             while (running.get()) {
+
+                // new connection
                 try (PgConnection pc = pgConnectionSupplier.get()) {
-                    boolean poll = true;
+                    connectionSetup(pc);
+
                     while (running.get()) {
-                        try (PreparedStatement ps = pc.prepareStatement(PgConstants.LISTEN_SQL)) {
-                            log.trace("Running LISTEN command");
-                            ps.execute();
-                        }
-                        if (poll) {
-                            // make sure, we did not miss anything while
-                            // reconnecting,
-                            postEvent("scheduled-poll");
-                            poll = false;
-                        }
-                        if (pgConnectionTester.test(pc)) {
-                            log.trace("Waiting for notifications for {}ms",
-                                    blockingWaitTimeInMillis);
-                            l.countDown();
-                            PGNotification[] notifications = pc.getNotifications(
-                                    blockingWaitTimeInMillis);
-                            if ((notifications != null) && (notifications.length > 0)) {
-                                final String name = notifications[0].getName();
-                                log.trace("notifying consumers for '{}'", name);
-                                postEvent(name);
-                            } else {
-                                log.trace("No notifications yet. Looping.");
-                            }
-                        } else {
-                            throw new SQLException("Connection is failing test");
-                        }
+                        PGNotification[] notifications = receiveNotifications(pc);
+                        informSubscriberOfChannelNotifications(notifications);
```

Comment:
this does not necessarily inform subscribers, but only if relevant notifications are in the array.
can we doe something like 

if (containsEventInserted(notifications))
  post(..

if think that'll make reading and reasoning about the code a bit easier.